packages/devtools_app/lib/src/screens/inspector_v2/inspector_controls.dart

```diff
@@ -0,0 +1,132 @@
+// Copyright 2024 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:devtools_app_shared/ui.dart';
+import 'package:flutter/material.dart';
+
+import '../../service/service_extension_widgets.dart';
+import '../../service/service_extensions.dart' as extensions;
+import '../../shared/analytics/constants.dart' as gac;
+import '../../shared/common_widgets.dart';
+import '../../shared/globals.dart';
+import 'inspector_controller.dart';
+import 'inspector_screen.dart';
+
+/// Control buttons for the inspector panel.
+class InspectorControls extends StatelessWidget {
+  const InspectorControls({
+    super.key,
+    required this.controller,
+  });
+
+  final InspectorController controller;
+
+  static const serviceExtensionButtonsIncludeTextWidth = 1200.0;
+
+  @override
+  Widget build(BuildContext context) {
+    return Row(
+      crossAxisAlignment: CrossAxisAlignment.start,
+      children: [
+        ValueListenableBuilder<bool>(
+          valueListenable: serviceConnection
+              .serviceManager.serviceExtensionManager
+              .hasServiceExtension(
+            extensions.toggleSelectWidgetMode.extension,
+          ),
+          builder: (_, selectModeSupported, __) {
+            return ServiceExtensionButtonGroup(
+              extensions: [
+                selectModeSupported
+                    ? extensions.toggleSelectWidgetMode
+                    : extensions.toggleOnDeviceWidgetInspector,
+              ],
+              minScreenWidthForTextBeforeScaling:
+                  InspectorScreenBodyState.minScreenWidthForTextBeforeScaling,
+            );
+          },
+        ),
+        const Spacer(),
+        HideImplementationWidgetsButton(controller: controller),
```

Comment:
nit: let's move this to the left button group with the select widget mode button. This should help with discoverability, especially since this will be directly above the tree that the setting modifies. We may also want to consider using a Switch, but perhaps we could get some UX guidance for that.

Reply:
Done!